File: examples/java/HelloPlugin/src/main/java/falcon/examples/hello/listeners/ChatListener.java
package falcon.examples.hello.listeners;

import falcon.api.Plugin;
import falcon.api.event.EventPriority;
import falcon.api.event.PlayerChatEvent;

public final class ChatListener {
    private ChatListener() {
    }

    public static void registerTo(Plugin plugin) {
        plugin.events().on(PlayerChatEvent.class, event -> {
            if (event.message().equals("ping")) {
                event.player().sendMessage("pong");
            }
        }, EventPriority.MONITOR);
    }
}

File: examples/dotnet/HelloPlugin/Listeners/ChatListener.cs
using Falcon;

namespace Hello.Listeners
{
    public static class ChatListener
    {
        public static void RegisterTo(Plugin plugin)
        {
            plugin.Events.On<PlayerChatEvent>(chatEvent =>
            {
                if (chatEvent.Message == "ping")
                {
                    chatEvent.Player.SendMessage("pong");
                }
            }, EventPriority.Monitor);
        }
    }
}
